internal/host/validator.go

```diff
@@ -491,12 +491,37 @@ func getNumEnabledHosts(hosts []*models.Host) int {
 	}
 	return ret
 }
+func (v *validator) belongsToL2MajorityGroup(c *validationContext, majorityGroups map[string][]strfmt.UUID) ValidationStatus {
+	if c.cluster.MachineNetworkCidr == "" {
+		return ValidationPending
+	}
+	return boolValue(funk.Contains(majorityGroups[c.cluster.MachineNetworkCidr], *c.host.ID))
+}
+
+func (v *validator) belongsToL3MajorityGroup(c *validationContext, majorityGroups map[string][]strfmt.UUID) ValidationStatus {
```

Comment:
What about unit tests?

Reply:
The changes I did already in the transition_test.go covered these changes.

Reply:
Added some